app/models/concerns/profile_searching.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+# Methods that are related to profile searching
+module ProfileSearching
+  extend ActiveSupport::Concern
+
+  included do
+    scoped_search on: %i[id name ref_id account_id compliance_threshold
+                         external parent_profile_id]
+    scoped_search relation: :hosts, on: :id, rename: :system_ids
+    scoped_search relation: :hosts, on: :name, rename: :system_names
+    scoped_search on: :has_test_results, ext_method: 'test_results?',
+                  only_explicit: true, operators: ['=']
+    scoped_search on: :canonical, ext_method: 'canonical?', only_explicit: true,
+                  operators: ['=']
```

Comment:
We should also be able to search by OS version here, similar to how the wizard works via graphql.

Reply:
Thanks Andrew. Yes it will definitely help us to identify uniquely a policy (not more than 1 entities returned) when searching by ref_id, external=False, and the RHEL major version.